app/Actions/Photo/DuplicateFinder.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+/**
+ * SPDX-License-Identifier: MIT
+ * Copyright (c) 2017-2018 Tobias Reich
+ * Copyright (c) 2018-2025 LycheeOrg.
+ */
+
+namespace App\Actions\Photo;
+
+use App\Exceptions\Internal\LycheeLogicException;
+use App\Exceptions\Internal\QueryBuilderException;
+use App\Models\Photo;
+use Illuminate\Database\Query\Builder;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Facades\DB;
+
+/**
+ * Look for duplicates in the database.
+ */
+class DuplicateFinder
+{
+	/**
+	 * Quickly count the number of duplicates candidates.
+	 *
+	 * @param bool $with_album_constraint    Requires the duplicates to be in the same album
+	 * @param bool $with_checksum_constraint Requires the duplicates to have the same checksum
+	 * @param bool $with_title_constraint    Requires the duplicates to have the same title
+	 *
+	 * @return int
+	 */
+	public function checkCount(
+		bool $with_album_constraint,
```

Comment:
naming is hard but i'd call these `$mustBeWithinSameAlbum`, `$mustBeSameChecksum`, `$mustBeSameTitle`

Reply:
Fixed [e04205d](https://github.com/LycheeOrg/Lychee/pull/2889/commits/e04205d7307acfd5b6ee0a73107bf7d3b25ae497).